resharper/resharper-unity/src/Yaml/Psi/UnitySceneProcessor.cs

```diff
@@ -0,0 +1,147 @@
+using System.Linq;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.Unity.Yaml.Psi.Caches;
+using JetBrains.ReSharper.Plugins.Unity.Yaml.Psi.Modules;
+using JetBrains.ReSharper.Plugins.Yaml.Psi;
+using JetBrains.ReSharper.Plugins.Yaml.Psi.Parsing;
+using JetBrains.ReSharper.Plugins.Yaml.Psi.Tree;
+using JetBrains.ReSharper.Psi.Files;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Yaml.Psi
+{
+    [SolutionComponent]
+    public class UnitySceneProcessor
+    {
+        private readonly UnityVersion myVersion;
+        private readonly MetaFileGuidCache myMetaFileGuidCache;
+        private readonly UnityExternalFilesModuleFactory myFactory;
+
+        public UnitySceneProcessor(UnityVersion version, MetaFileGuidCache metaFileGuidCache, UnityExternalFilesModuleFactory factory)
+        {
+            myVersion = version;
+            myMetaFileGuidCache = metaFileGuidCache;
+            myFactory = factory;
+        }
+
+        private static bool IsStripped(IYamlDocument element)
+        {
+            return ((element.Body.BlockNode as IChameleonBlockMappingNode)?.Properties?.LastChild as YamlTokenType.GenericTokenElement)?
```

Comment:
Use `IBlockMappingNode` instead of `IChameleonBlockMappingNode` - this is mostly a marker interface.